packages/@react-aria/table/src/useTable.ts

```diff
@@ -109,7 +111,48 @@ export function useTable<T>(props: TableProps<T>, state: TableState<T>, ref: Ref
     return sortDirection && column ? formatMessage(`${sortDirection}Sort`, {columnName}) : undefined;
   }, [sortDirection, column, state.collection.columns]);
 
-  let descriptionProps = useDescription(sortDescription);
+
+  // do we have something that might be better suited for this? issue is that useInteractionModality
+  // doesn't return quite enough information
+  let [interactionType, setInteractionType] = useState('');
+  useEffect(() => {
+    let onPointerDown = (e) => {
+      setInteractionType(e.pointerType);
+    };
+    let onKeyDown = () => {
+      setInteractionType('keyboard');
+    };
+    let tableRef = ref.current;
+    tableRef.addEventListener('pointerdown', onPointerDown, true);
+    tableRef.addEventListener('pointerup', onPointerDown, true);
+    tableRef.addEventListener('keydown', onKeyDown, true);
+    tableRef.addEventListener('keyup', onKeyDown, true);
+    return () => {
+      tableRef.removeEventListener('pointerdown', onPointerDown, true);
+      tableRef.removeEventListener('pointerup', onPointerDown, true);
+      tableRef.removeEventListener('keydown', onKeyDown, true);
+      tableRef.removeEventListener('keyup', onKeyDown, true);
+    };
+  });
+  let interactionDescription = useMemo(() => {
+    let selectionMode = state.selectionManager.selectionMode;
+    let selectionBehavior = state.selectionManager.selectionBehavior;
+    // if we're in replace but can select multiple, then when using touch it's long press to enter selection mode
+    // if we can't tell what mode we're in, then we're probably in voice over, in which case we need the description to be set
+    // before the user interacts so they know they can enter selection mode before they actually have to interact
+    let message = formatMessage('longPressToSelect');
+    if (interactionType === 'touch') {
+      message = formatMessage('longPressToSelect');
+    }
+    return selectionBehavior === 'replace' && selectionMode !== 'none' && props.onAction ? message : undefined;
```

Comment:
Why is this only replace? Isn't this for long press which is both replace and toggle?

Reply:
the message only applies to get into selection mode, which only happens when we're in replace mode, see https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/selection/src/useSelectableItem.ts#L139 as well